vernac/vernacinterp.ml

```diff
@@ -106,9 +106,14 @@ and vernac_load ~verbosely entries =
   stack, pm
 
 and interp_control ~st ({ CAst.v = cmd; loc }) =
-  interp_control_gen ~loc ~st cmd.control
-    ~unfreeze_transient:Vernacstate.Synterp.unfreeze
-    (fun () -> interp_expr ?loc ~st cmd)
+  Util.try_finally (fun () ->
+      Loc.set_current_command_loc loc;
```

Comment:
We keep the current command loc in a global ref (put in loc.ml because I had no better idea) to be a backup when a command does not pass more precise loc info to declare.

Reply:
Can we please instead fix the commands to pass info as done in #16261 may provide some more places to fix (I didn't check tho).

The way I see it is that the hack is a recipe for problems, on the other hand, if a command didn't send its location, it can be fixed easily.

Reply:
Not necessarily easy, typically `abstract` has no good location at hand

also I don't see the problem so don't want to spend much time removing it

Reply:
Abstract could use the location of the tactic, couldn't it?

Actually if the hack was more local it would be OK, but the way it is is IMO problematic. If we put `loc` in a global variable, why for example not to put poly, and other arguments to cinfo in global variables too?

Design-wise, these are the main problems I see:
- the global variable may look fine today, but in the end, it will create maintenance problems if for example `vernacinterp` is refactored in a way more friendly for Flèche
- `vernacinterp` knows nothing about the command at hand, so what the PR is doing now is an _approximation_ of the location; this is limiting for LSP for example (see the current jump to def protocol spec)
- moreover I don't think fixing this hack should take a lot of time, as the previous PR attempting this shows

Reply:
>Abstract could use the location of the tactic, couldn't it?

no, the tactic may be in some `Ltac` definition

Reply:
Indeed to fix this properly (IMO) would require a lot of refactoring that indeed shouldn't be done here, tho if we ever reach to the point to allow DMs to replace vernacinterp as to take advantage of the data deps there, we could indeed implement this _à la Lean_ easily.

IMO we can keep the hack for abstract (and call the variable `command_loc_for_abstract` of something like that)

Note how the location will still be imprecise in the case of chained tactics), but I think we should remove all the other uses of the hack.

Reply:
It's not just for abstract, it's for any command whose implementor has other things to do than pass around locs

Reply:
Actually locs are pretty important IMO, so command implementors should care about them, but if you wanna go this way, OK then.

Reply:
>how would the developer discover that they need to override the global ref in order to pass the loc, if they want to override the loc?

They shouldn't override the loc, it's "current command loc" not "current whatever loc"
If they want a more precise loc the ?loc argument needs to be added to the api

Reply:
Indeed, my worry is that how people looking at the API of commands will discover that this is handled in a global variable. Anyways IMHO I'm happy to end the current discussion, I wrote many years ago a CEP to have a more common vernac handling, this actually has uses in UIs, but not yet. We can retake this discussion if we need on the CEP.